src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -475,6 +519,254 @@ class DefinerSegment(BaseSegment):
+                ),
+            ),
+        ),
+    )
+
+
+# I wanted to do an override and just insert the new Refs but I need
+# to have the IntoClause be before FromClauseSegment, but it doesn't work to add the `before` clause
+# looking for suggestions on how to do this differently, if possible
+@mysql_dialect.segment(replace=True)
+class UnorderedSelectStatementSegment(BaseSegment):
+    """A `SELECT` statement without any ORDER clauses or later.
+    This is designed for use in the context of set operations,
+    for other use cases, we should use the main
+    SelectStatementSegment.
+    """
+
+    type = "select_statement"
+    # match grammar. This one makes sense in the context of knowing that it's
+    # definitely a statement, we just don't know what type yet.
+    match_grammar = StartsWith(
+        # In mysql, the select clause may include an INTO statement
+        # to assign values from columns/functions to corresponding
+        # local or session variables
+        Ref("SelectClauseSegment"),
+        terminator=OneOf(
+            Ref("IntoClauseSegment"),
+            Ref("SetOperatorSegment"),
+            Ref("WithNoSchemaBindingClauseSegment"),
+            Ref("OrderByClauseSegment"),
+            Ref("LimitClauseSegment"),
+            Ref("NamedWindowSegment"),
+            Ref("IndexHintClauseSegment"),
+            Ref("ForClauseSegment"),
+        ),
+        enforce_whitespace_preceeding_terminator=True,
+    )
+
+    parse_grammar = Sequence(
+        Ref("SelectClauseSegment"),
+        # Dedent for the indent in the select clause.
+        # It's here so that it can come AFTER any whitespace.
+        Dedent,
+        Ref("IntoClauseSegment", optional=True),
+        Ref("FromClauseSegment", optional=True),
+        Ref("IndexHintClauseSegment", optional=True),
+        Ref("WhereClauseSegment", optional=True),
+        Ref("GroupByClauseSegment", optional=True),
+        Ref("HavingClauseSegment", optional=True),
+        Ref("ForClauseSegment", optional=True),
+    )
+
+
+@mysql_dialect.segment(replace=True)
+class SelectClauseElementSegment(BaseSegment):
+
+    """An element in the targets of a select statement."""
+
+    type = "select_clause_element"
+
+    match_grammar = ansi_dialect.get_segment(
+        "SelectClauseElementSegment"
+    ).match_grammar.copy()
+
+    parse_grammar = ansi_dialect.get_segment(
+        "SelectClauseElementSegment"
+    ).parse_grammar.copy()
+
+
+# I am unclear why I have to override this segement, but if I don't then new segments won't parse
+# looking for suggestions on how to avoid this since it seems unnecessary
+@mysql_dialect.segment(replace=True)
+class SelectClauseSegment(BaseSegment):
+    """A group of elements in a select target statement."""
+
+    type = "select_clause"
+    match_grammar = StartsWith(
+        Sequence("SELECT", Ref("WildcardExpressionSegment", optional=True)),
+        terminator=OneOf(
+            "INTO",
```

Comment:
While you shouldn't have to override the entire segment, I see that you have added `"INTO"` here -- this is not present in the ANSI dialect. So I think you'll be able to use `copy()` for fields where MySQL is identical to ANSI, but if it's different, you'll have to either replicate some of the code or use some of the special features of `.copy()` that let you modify the copy.